### PR TITLE
Atomization #14403 Vali harvester sprites

### PR DESCRIPTION
**Pull request by Xander3359**
## About The Pull Request
This is an atomization so maintainers can easily merge just the sprite + code changes.

General changes:
- Harvester sprites will now visibly show which reagent is loaded. Both item + mob sprites are affected
image
From top to bottom: Bicaridine, Kelotane, Tramadol, Tricordrazine, Empty

Code changes:
- Moved harvester weapons + sprites to their own files
- Moved get_mechanics_info to the harvester component
## Why It's Good For The Game
- Making the chemicals visible means that you have a visual indicator of what will happen when you get smacked by a harvester.
- Code is nicer when it's in its own file

## Changelog
:cl:
imageadd: Harvesters will now have a visual based on whichever reagent is loaded
code: Harvesters now have sprites + code in their own separate file
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/harvester.dm [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+	name = "\improper HP-S Harvester blade"
+	desc = "TerraGov Marine Corps' experimental High Point-Singularity 'Harvester' blade. An advanced weapon that trades sheer force for the ability to apply a variety of debilitating effects when loaded with certain reagents. Activate after loading to prime a single use of an effect. It also harvests substances from alien lifeforms it strikes when connected to the Vali system."
+	icon_state = "vali_sword"
+	//item_state = "vali_sword"
+	force = 60
+	attack_speed = 12
+
+/obj/item/weapon/harvester/sword/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/harvester)
```
Comment (psykzz) [inline code/game/objects/items/weapons/harvester.dm:54]: Is there every a `/obj/item/weapon/harvester` that doesn't get the harvester component?

Comment (Xander3359) [reply code/game/objects/items/weapons/harvester.dm:54]: The claymore has different args ``AddComponent(/datum/component/harvester, 60, TRUE)``

Comment (psykzz) [reply code/game/objects/items/weapons/harvester.dm:54]: Then use like `LateInit` and override it in the claymore

File: code/datums/components/harvester.dm [hunk 2]
```diff
@@ -156,6 +166,27 @@
...
+///Updates the color of the weapon based on what chem is loaded in. Updates both the item sprite and mob sprite
+/datum/component/harvester/proc/update_loaded_color(datum/source)
+	SIGNAL_HANDLER
+	var/obj/item/item_parent = parent
+	var/image/item_overlay = image('icons/obj/items/vali.dmi', item_parent, "[initial(item_parent.icon_state)]_loaded")
+	if(!loaded_reagent)
+		item_overlay.color = COLOR_GREEN
+	else
+		item_overlay.color = initial(loaded_reagent.color)
+	item_parent.overlays.Add(item_overlay)
```
Comment (Lumipharon) [inline code/datums/components/harvester.dm:178]: you're misusing this signal/applying overlays incorrectly.

Look at where this signal is coming from, you need to add the overlay to the list its sending as an arg, not directly add it.

File: code/datums/components/harvester.dm [hunk 3]
```diff
@@ -156,6 +166,27 @@
 		target.adjustStaminaLoss(-30)
 		target.heal_overall_damage(6, 0, updating_health = TRUE)
 
+///Updates the color of the weapon based on what chem is loaded in. Updates both the item sprite and mob sprite
```
Comment (Lumipharon) [inline code/datums/components/harvester.dm:169]: incorrect comment

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/harvester.dm [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+	sharp = IS_SHARP_ITEM_ACCURATE
+	hitsound = 'sound/weapons/slash.ogg'
+
+/obj/item/weapon/harvester/knife/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/harvester, 5)
+
+/obj/item/weapon/harvester/knife/equipped(mob/user, slot)
+	. = ..()
+	toggle_item_bump_attack(user, FALSE)
```
[the remark this answers is not in the document]
Comment (Xander3359) [reply code/game/objects/items/weapons/harvester.dm:85]: Huh, I guess vali knife was meant to have bump slash, but whoever coded it in the first place made a mistake...

**Comment by psykzz** [comment]
PR title should be updated to describe the changes

File: code/modules/codex/codex_atom.dm [hunk 5]
```diff
@@ -13,6 +13,9 @@
 	return entry
 
 /atom/proc/get_mechanics_info()
+	var/list/mechanics_text = list()
```
Comment (Lumipharon) [inline code/modules/codex/codex_atom.dm:16]: this should probs call parent, but that can wait for another pr

**Review by Lumipharon — APPROVED** [review]

**Comment by Lumipharon** [comment]
mix the arg stuff and I'll merge

**Pull request merged**